### PR TITLE
Lava Boss Mob

### PR DESCRIPTION
**Pull request by RemieRichards**
Adds an extremely strong "Boss" Mob, themed around Lava:
- 1000 HP
- Heals from all forms of fire/explosions/burn damage
- Has 3 Unique Animated attacks: Groundsmash, Firewalls and Lavarivers
- Standard Hostile mob AI with some code to make it trigger the 3 abilities the more it gets hurt
- Has Action buttons to allow it to be controlled by players

Adds a helper to pick 3 random points on the edge of a square, used for Lavariver power

Action Buttons:
Imgur

Groundsmash (it has screen shake too, but old gif):
Imgur

Lava rivers + Firewalls:
Imgur

### CODE REVIEW

**Comment by Nienhaus** [comment]
Holy shit, it almost wants me to make new fire in a similar way to bomber man.

**Comment by Zilenan91** [comment]
Isn't it spawning lava everywhere super overpowered since it's literally an instant kill unless you have a fire extinguisher

**Comment by RemieRichards** [comment]
@Zilenan91 that's the rarest power, the gif is me controlling it with no delays set.

**Comment by GunHog** [comment]
Make this its theme song: https://www.youtube.com/watch?v=lC7QiUAhv3Y

**Comment by RemieRichards** [comment]
Oh uh yeah sounds would be much appreciated.
Lava bubbles, roars, flame sounds, ground smash etc. @Cuboos

**Comment by PKPenguin321** [comment]
hella dope

**Comment by Cuboos** [comment]
I've been a little busy lately, but when i can get around to it, i'll see what i can do.

File: code/modules/mob/living/simple_animal/hostile/lavaboss.dm [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+
+
+
+/obj/structure/flora/rock/lava_attack_rock/New()
+	..()
+	spawn(10)
+		var/turf/T = get_turf(src)
+		for(var/atom/movable/AM in T.contents)
+			AM.ex_act(1)
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/hostile/lavaboss.dm]: ex_ac(1) gibs human mobs I think you want ex_act(3)

**Comment by Niknakflak** [comment]
Yea codersprite.  Nienhaus or someone, can you sprite something that isn't bleh?

**Comment by Shadowlight213** [comment]
Codersprite looks like lizard prime or something.

**Comment by Dorsisdwarf** [comment]
post it as a png and I'll take a crack

**Comment by Bawhoppen** [comment]
>coder sprite 


>better than like half of our actual sprites

**Comment by GunHog** [comment]
It is actually decently done. RR is a decent spriter, too.

**Comment by Niknakflak** [comment]
It's head is all one color, zero shading

**Comment by RemieRichards** [comment]
incorrect, the beard/thing is shaded.

**Comment by Niknakflak** [comment]
You know what I mean, the top of the head looks odd because it's a single color, now that I'm off my phone I do see the beard shading.

**Comment by Steelpoint** [comment]
Its a nice looking sprite, but it reminds me more of a cult construct.

**Comment by Ausops** [comment]
>animated

brave

**Comment by Nienhaus** [comment]
Okay I do kinda any to try something now.

File: code/modules/mob/living/simple_animal/hostile/lavaboss.dm [hunk 2]
```diff
@@ -0,0 +1,262 @@
+
+#define NOT_ANGRY 0
+#define ANNOYED 250
+#define ANGRY	500
+#define PISSED	750
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/hostile/lavaboss.dm]: lol these defines

**Comment by optimumtact** [comment]
man this is the **HOTTEST** change of the year

**Comment by KorPhaeron** [comment]
Is there any chance of making this a /mob/living/simple_animal/boss/lavaboss and having all the anger behaviours part of the boss class?

I'd like to do similar stuff for other large mobs so having it be all modular would be nice. I mean if not I guess I'll be able to do it later

>faction = list("lavaland")

oh shit its canon now I gotta finish it

File: code/modules/mob/living/simple_animal/hostile/lavaboss.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/mob/living/simple_animal/hostile/lavaboss.dm]: where's "spilled all my coffee" define

**Comment by GunHog** [comment]
The alien queen is like a spiderling compared to this beast.

**Comment by Ausops** [comment]
as a suggestion, move the sprite up as if it's floating when it raises it's claws/feet then crash it right back down one or two pixel or two and immediately move it back to it's starting position when the sprite resets, if possible.

**Comment by RemieRichards** [comment]
@KorPhaeron I could move that to an abstract boss class sure, I didn't by default because it's largely just a number being tracked and then whatever the dev wants the mob to react to the number, I'll do it when I get home.

Also not sure if people realised, but anger doesn't go down when a creature is healed, meaning other bosses (maybe this one if I add some more switches/make some maths) get mad and STAY mad, so try to finish the job if you weaken one!

**Comment by optimumtact** [comment]
update to latest master
git rebase -Xignore-all-whitespace upstream/master

**Comment by WJohn** [comment]
Any word on this?

**Comment by Zilenan91** [comment]
it's dead, jim

**Comment by RemieRichards** [comment]
@WJohn 
@Zilenan91 

hardest part of this is/was the sprites, which are done
the code is VERY simple shit.

this PR isn't merge-ready for the same reason basil is down.
Laziness (that and I want to redo the boss logic)

**Comment by GunHog** [comment]
Remember that RR's laziness is <b>LEGENDARY</b>, as it utterly kills some of her most hyped features!!

**Comment by optimumtact** [comment]
i killed it

**Comment by RemieRichards** [comment]
My laziness is only partially innate.
a lot of it stems from the fact that I have emotional issues, mentioning repeatedly that I fail to complete things... really doesn't help, it's only funny the first few times.
My drive to do things is very much all over the place, that and given coding is one of the few things I do decently, I've become a bit of a perfectionist, a LOT of things don't make the cut because of that.

@optimumtact nah, you didn't, as I said, Laziness did, I've already ran the command on this PR and everything, just didn't do the actual conflict-fix part.

bit heavy, but there you go.

**Comment by Cuboos** [comment]
So should i still do sounds or what?

**Comment by Zilenan91** [comment]
probably, just to keep progress rolling.

**Comment by RemieRichards** [comment]
@Cuboos yes, please.

**Comment by tkdrg** [comment]
@RemieRichards

**Comment by Bawhoppen** [comment]
#stillaterrariaboss

**Comment by tkdrg** [comment]
Conflicts fixed now, all that's left is for Remie to fix the boss AI.

File: code/modules/mob/living/simple_animal/hostile/lavaboss.dm [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+
+/datum/action/lavaboss/Trigger()
+	if(..())
+		if(boss)
+			if(boss.next_ability < world.time)
+				return 1
+		return 0
+
+/datum/action/avaboss/groundsmash
```
Comment (MatthewWork) [inline code/modules/mob/living/simple_animal/hostile/lavaboss.dm]: DANGER DANGER

**Comment by Razharas** [comment]
Merge conflicts are dead, heil to the merge conflicts

**Comment by Cuboos** [comment]
https://dl.dropboxusercontent.com/u/9654207/Lava%20Monster/FireWall.ogg
https://dl.dropboxusercontent.com/u/9654207/Lava%20Monster/GroundSmash.ogg
https://dl.dropboxusercontent.com/u/9654207/Lava%20Monster/Lava_river.ogg

Let me know if you need anything else or if anything needs to be changed.

**Comment by Dorsisdwarf** [comment]
@RemieRichards reeeembly reeeeeeemblyyyy

**Comment by Shadowlight213** [comment]
@RemieRichards boop

**Comment by RemieRichards** [comment]
@Shadowlight213 beep

**Comment by duncathan** [comment]
I believe in you

**Comment by Cuboos** [comment]
@RemieRichards Well Remie, do you like the sounds or not?

**Comment by RemieRichards** [comment]
@Cuboos 
lot of dead air in groundsmash

but yes I like them.

**Comment by tkdrg** [comment]
as much as it saddens me, this is stale, so i'm closing it in a futile attempt to declutter the PR list
don't feel discouraged please, this is still something I'd love to merge if you decide to revisit it

**Comment by KazeEspada** [comment]
RemieneverfinishanythingRichards

**Pull request closed without merging**